File: app/src/main/java/com/example/pam_project/features/tasks/list/TaskInformation.java
package com.example.pam_project.features.tasks.list;

import com.example.pam_project.utils.constants.TaskStatus;

public class TaskInformation implements Comparable<TaskInformation> {
    private final String title;
    private final String description;
    private final boolean isUrgent;
    private final TaskStatus status;
    private long id;

    public TaskInformation(final long id, final String title, final String description, final boolean isUrgent, final TaskStatus status) {
        this.id = id;
        this.title = title;
        this.description = description;
        this.isUrgent = isUrgent;
        this.status = status;
    }

    public TaskInformation(final String title, final String description, final boolean isUrgent, final TaskStatus status) {
        this.title = title;
        this.description = description;
        this.isUrgent = isUrgent;
        this.status = status;
    }

    public String getTitle() {
        return title;
    }

    public long getId() {
        return id;
    }

    public String getDescription() {
        return description;
    }

    public boolean getUrgency() {
        return isUrgent;
    }

    public TaskStatus getStatus() {
        return status;
    }

    @Override
    public boolean equals(Object o) {
        if (o == null)
            return false;
        if (!(o instanceof TaskInformation))
            return false;
        TaskInformation other = (TaskInformation) o;
        return id == other.getId() && title.equals(other.getTitle()) &&
                description.equals(other.getDescription()) && isUrgent == other.getUrgency() &&
                status.equals(other.getStatus());
    }

    @Override
    public int compareTo(TaskInformation o) {
        return this.getStatus().compareTo(o.getStatus());
    }
}

File: app/src/main/java/com/example/pam_project/features/tasks/list/TaskInformation.kt
package com.example.pam_project.features.tasks.list

import com.example.pam_project.utils.constants.TaskStatus

class TaskInformation(id: Long, title: String?, description: String?, isUrgent: Boolean,
                      status: TaskStatus?) : Comparable<TaskInformation> {

    var title: String? = title
        private set
    var description: String? = description
        private set
    var urgency: Boolean = isUrgent
        private set
    var status: TaskStatus? = status
        private set
    var id: Long = id
        private set

    override fun equals(other: Any?): Boolean {
        if (other == null) return false
        if (other !is TaskInformation) return false
        return id == other.id && title == other.title && description == other.description
                && urgency == other.urgency && status == other.status
    }

    override fun compareTo(other: TaskInformation): Int {
        return status!!.compareTo(other.status!!)
    }

    override fun hashCode(): Int {
        var result = title?.hashCode() ?: 0
        result = 31 * result + (description?.hashCode() ?: 0)
        result = 31 * result + urgency.hashCode()
        result = 31 * result + (status?.hashCode() ?: 0)
        result = 31 * result + id.hashCode()
        return result
    }
}